src/dcap_provider.cpp

```diff
@@ -453,11 +484,23 @@ extern "C" quote3_error_t sgx_ql_get_quote_config(
 
     try
     {
-        auto p_quote_config = std::make_unique<sgx_ql_config_t>();
-        memset(p_quote_config.get(), 0, sizeof(*p_quote_config));
-        p_quote_config->version = SGX_QL_CONFIG_VERSION_1;
-
         const std::string cert_url = build_pck_cert_url(*p_pck_cert_id);
+
+        if (auto cache_hit = local_cache_get(cert_url))
+        {
+            // we could optimize out a memory allocation and copy here,
+            // but it's subtly tricky because it means we'd need to ensure
+            // the

```

Comment:
"ensure the …?"

Reply:
haha, @sethmoo wrote this comment. I can remove it.